tt_metal/tools/profiler/noc_event_profiler_utils.hpp

```
@@ -92,6 +94,66 @@ inline void dumpClusterCoordinatesAsJson(const std::filesystem::path& filepath)
+                fabric_node_id.mesh_id.get(), fabric_node_id.chip_id};
+
+            for (const auto& direction : tt::tt_fabric::FabricContext::routing_directions) {
+                auto eth_routing_planes_in_dir =
+                    tt::tt_fabric::get_active_fabric_eth_routing_planes_in_direction(fabric_node_id, direction);
+
+                while (device_routing_planes.size() < eth_routing_planes_in_dir.size()) {
+                    device_routing_planes.push_back(
+                        {{"routing_plane_id", device_routing_planes.size()},
+                         {"ethernet_channels", nlohmann::ordered_json::object()}});
+                }
+
+                for (int j = 0; j < eth_routing_planes_in_dir.size(); j++) {
+                    chip_id_t eth_channel = eth_routing_planes_in_dir[j];
+                    device_routing_planes[j]["ethernet_channels"][magic_enum::enum_name(direction)] = eth_channel;
+                }
+            }
+
+            topology_json["routing_planes"].push_back(
+                {{"device_id", physical_chip_id}, {"device_routing_planes", device_routing_planes}});
+        }
+    }
+
+    topology_json["eth_chan_to_coord"] = nlohmann::ordered_json::object();
+    auto physical_chip_id = *(cluster.get_cluster_desc()->get_all_chips().begin());
+    for (int j = 0; j < cluster.get_soc_desc(physical_chip_id).get_num_eth_channels(); j++) {
+        tt::umd::CoreCoord edm_eth_core =
+            cluster.get_soc_desc(physical_chip_id).get_eth_core_for_channel(j, CoordSystem::PHYSICAL);
+        topology_json["eth_chan_to_coord"][std::to_string(j)] = {edm_eth_core.x, edm_eth_core.y};
+    }
+
+    std::ofstream topology_json_ofs(filepath);
+    if (topology_json_ofs.is_open()) {
+        topology_json_ofs << topology_json.dump(2);
+    } else {
+        log_error(tt::LogMetal, "Failed to open file '{}' for dumping topology", filepath.string());
```

I think we need to throw an exception here? Add `TT_FATAL(topology_json_ofs.is_open(), "Failed to open file...`

---

Should we be terminating the entire program if we're unable to open a file the device profiler uses (i.e. it doesn't affect the main execution)?

---

When / how do you call this function?